sdk/logs/src/main/java/io/opentelemetry/sdk/logs/SdkLogBuilder.java

```diff
@@ -17,10 +18,12 @@
 
   private final LogDataBuilder logDataBuilder;
   private final LogEmitterSharedState logEmitterSharedState;
+  private final LogLimits logLimits;
 
   SdkLogBuilder(LogEmitterSharedState logEmitterSharedState, LogDataBuilder logDataBuilder) {
     this.logEmitterSharedState = logEmitterSharedState;
     this.logDataBuilder = logDataBuilder;
+    this.logLimits = logEmitterSharedState.getLogLimits();
```

Comment:
Is it important to hold this reference? Does it limit the utility of the `Supplier` in the shared state then, or is the really the intention? (I probably don't understand the builder's lifecycle well enough).

Reply:
We need to have predictable semantics about when the log limits are resolved. Spans have the same concept, and `Supplier<SpanLimits>` is invoked when `Tracer.spanBuilder(String spanName)` is called. I think it makes sense for the log limits to resolved when the log builder is initialized, versus when attributes are first added.